### PR TITLE
options: add --secondary-sub-ass-override

### PR DESCRIPTION
**Pull request by dyphire**
Follow up to some discussion we had in https://github.com/mpv-player/mpv/pull/13073.
Regardless of whether we implement fully configurable for the secondary subs options in the future, the `ass_style_override` needs to be handled separately.

### CODE REVIEW

File: sub/dec_sub.c [hunk 1]
```diff
@@ -149,7 +149,7 @@ static struct sd *init_decoder(struct dec_sub *sub)
             .preload_ok = true,
         };
 
-        if (sd->driver->init(sd) >= 0)
+        if (sd->driver->init(sd, sub->order) >= 0)
```
Comment (Dudemanguy) [inline sub/dec_sub.c:152]: Instead of passing `sub->order` everywhere, it's probably easier to just add `order` to the `sd` struct at this point.

Comment (dyphire) [reply sub/dec_sub.c]: Done.

File: DOCS/man/options.rst [hunk 2]
```diff
@@ -2499,6 +2499,12 @@ Subtitles
     This also controls some bitmap subtitle overrides, as well as HTML tags in
     formats like SRT, despite the name of the option.
 
+``--secondary-sub-ass-override=<yes|no|force|scale|strip>``
+    Control whether user secondary substyle overrides should be applied. This
+    is similar to ``--sub-ass-override`` but for secondary subtitles.
+
+    Default: strip.
```
Comment (Dudemanguy) [inline DOCS/man/options.rst:2506]: I know that this is just preserving the old behavior but is there any reason to not just default to `yes` now? We already got rid of a hardcoded alignment value for secondary subs so it's technically different anyway.

Comment (dyphire) [reply DOCS/man/options.rst:2506]: As mentioned in https://github.com/mpv-player/mpv/pull/13073#issuecomment-1851930198, if we default to retaining the secondary subtitle style, using sub subtitles when there are two ass subtitles and their dialogue subtitles have a pos style will cause them to overlap.

I also have two screenshots here to show the errors encountered in rendering bilingual ass subtitles while retaining the secondary subtitle style.
images: 


I think retaining old behaviors can better handle some special situations.

Comment (Dudemanguy) [reply DOCS/man/options.rst:2506]: Sure sounds reasonable.

**Comment by dyphire** [comment]
After carefully reviewing https://github.com/mpv-player/mpv/pull/13073 again, I found that it cannot be closed temporarily, so I updated the commit information.

**Review by Dudemanguy — APPROVED** [review]
Works fine.

File: DOCS/man/options.rst [hunk 3]
```diff
@@ -2499,6 +2499,12 @@ Subtitles
     This also controls some bitmap subtitle overrides, as well as HTML tags in
     formats like SRT, despite the name of the option.
 
+``--secondary-sub-ass-override=<yes|no|force|scale|strip>``
+    Control whether user secondary substyle overrides should be applied. This
+    is similar to ``--sub-ass-override`` but for secondary subtitles.
```
Comment (sfan5) [inline DOCS/man/options.rst:2504]:
```suggestion
    Control whether user secondary substyle overrides should be applied. This
    works exactly like ``--sub-ass-override``.
```

**Pull request merged**